src/main/java/com/google/devtools/build/lib/remote/AbstractActionInputPrefetcher.java

```diff
@@ -182,10 +218,15 @@ private Completable prefetchInputTree(
                 () -> {
                   HashSet<Path> dirs = new HashSet<>();
 
-                  // Tree root is created by Bazel before action execution, but the permission is
-                  // changed to 0555 afterwards. We need to set it as writable in order to move
-                  // files into it.
-                  treeRoot.setWritable(true);
+                  if (!treeRoot.exists()) {
+                    // If the tree root doesn't already exist, create it now. This can happen if
+                    // we're prefetching to a different location and symlinking into it.
+                    treeRoot.createDirectory();
```

Comment:
Consider using [createWritableDirectory](https://github.com/bazelbuild/bazel/blob/d44d4ea960fdc7260977e69b530e053baebf7fb8/src/main/java/com/google/devtools/build/lib/vfs/Path.java#L436-L448) to get it all done in 1 call.

Reply:
Done.